### PR TITLE
mailchimp stuff

### PR DESCRIPTION
**Pull request by Faiz**
## What does this PR do?
Added a management command to add existing devs to the mailing list.
Then any new dev who signs up will automatically be added to the list.

Needs two env variables setup in the .env
MAILCHIMP_ENDPOINT and MAILCHIMP_API_KEY

### CODE REVIEW

**Comment by ChristopherHammond13** [comment]
Build finished. 
Jenkins statics successfully built.

**Comment by ChristopherHammond13** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.uclapi.com/job/UCLAPI/112/
Jenkins statics successfully built.

**Comment by ChristopherHammond13** [comment]
Build finished. 
Jenkins statics successfully built.

**Comment by ChristopherHammond13** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.uclapi.com/job/UCLAPI/113/
Jenkins statics successfully built.

**Comment by ChristopherHammond13** [comment]
Build finished. 
Jenkins statics successfully built.

**Comment by ChristopherHammond13** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.uclapi.com/job/UCLAPI/114/
Jenkins statics successfully built.

File: backend/uclapi/dashboard/management/commands/add_existing_people_to_mailing_list.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from django.core.management.base import BaseCommand
+
+from dashboard.models import User
+from dashboard.api_applications import add_user_to_mailing_list
+
+
+class Command(BaseCommand):
+
+    help = 'creates the lock used to decide which bucket to flush and fill'
```
Comment (wilhelmklopp) [inline backend/uclapi/dashboard/management/commands/add_existing_people_to_mailing_list.py]: could you update this to describe what this management command does

File: backend/uclapi/dashboard/management/commands/add_existing_people_to_mailing_list.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+    def handle(self, *args, **options):
+        # delete existing locks
```
Comment (wilhelmklopp) [inline backend/uclapi/dashboard/management/commands/add_existing_people_to_mailing_list.py]: this comment seems like a leftover

File: backend/uclapi/dashboard/views.py [hunk 3]
```diff
@@ -43,6 +44,8 @@ def shibboleth_callback(request):
         )
 
         new_user.save()
+        add_user_to_mailing_list(new_user.email, new_user.given_name)
```
Comment (wilhelmklopp) [inline backend/uclapi/dashboard/views.py]: could you make this display name instead of given name?

Comment (Faiz) [reply backend/uclapi/dashboard/views.py]: whats display name?

Comment (Faiz) [reply backend/uclapi/dashboard/views.py]: oh full name. cool

**Comment by ChristopherHammond13** [comment]
Build finished. 
Jenkins statics successfully built.

**Comment by ChristopherHammond13** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.uclapi.com/job/UCLAPI/115/
Jenkins statics successfully built.

**Comment by ChristopherHammond13** [comment]
Build finished. 
Jenkins statics successfully built.

**Comment by ChristopherHammond13** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.uclapi.com/job/UCLAPI/124/
Jenkins statics successfully built.

**Comment by ChristopherHammond13** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.uclapi.com/job/UCLAPI/130/
Jenkins statics successfully built.

**Comment by ChristopherHammond13** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.uclapi.com/job/UCLAPI/131/
Jenkins statics successfully built.

**Comment by ChristopherHammond13** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.uclapi.com/job/UCLAPI/132/
Jenkins statics successfully built.

**Comment by ChristopherHammond13** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.uclapi.com/job/UCLAPI/133/
Jenkins statics successfully built.

**Comment by ChristopherHammond13** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.uclapi.com/job/UCLAPI/148/
Jenkins statics successfully built.

**Pull request merged**